experimental/fs/fs.go

```diff
@@ -0,0 +1,35 @@
+package fs
+
+import (
+	"context"
+
+	"github.com/tetratelabs/wazero"
+	"github.com/tetratelabs/wazero/api"
+	"github.com/tetratelabs/wazero/internal/wasm"
+)
+
+// FSKey is a context.Context Value key. It allows overriding fs.FS for WASI.
+//
+// See https://github.com/tetratelabs/wazero/issues/491
+type FSKey struct{}
+
+// Closer converts wasm.FSContext to api.Closer
+type Closer struct {
```

Comment:
probably after changing FSContext.Close you don't need this type anymore (if you did, it should be not exported)

Reply:
Thanks for confirming. I just didn't want to change the existing signature. Fixed https://github.com/tetratelabs/wazero/pull/571/commits/1aea8ea87c9cee34d33c392497348722732ce6be.